### PR TITLE
Addresses #670  don't add e columns for excluded fields

### PR DESCRIPTION
**Pull request by hlgp**

### CODE REVIEW

File: warehouse/ingest-core/src/main/java/datawave/ingest/metadata/EventMetadata.java [hunk 1]
```diff
@@ -216,6 +216,11 @@ protected void addEventField(IngestHelperInterface helper, RawRecordContainer ev
             return;
         }
         
+        if (helper.isShardExcluded(fieldName)) {
+            log.debug("{} is an excluded field, not adding to event", fieldName);
```
Comment (alerman) [inline warehouse/ingest-core/src/main/java/datawave/ingest/metadata/EventMetadata.java]: Should this message be "not adding to event **fields**"?

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java [hunk 2]
```diff
@@ -551,6 +551,11 @@ public void addIndexOnlyField(String fieldName) {
         indexOnlyFields.add(fieldName);
     }
     
+    @Override
+    public void addShardExclusionField(String fieldName) {
```
Comment (ivakegg) [inline warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java]: Lets use this method around line 393

File: warehouse/ingest-core/src/main/java/datawave/ingest/metadata/EventMetadata.java (hunk 1 above)
Comment (hlgp) [reply warehouse/ingest-core/src/main/java/datawave/ingest/metadata/EventMetadata.java]: no, it should say "not adding event or frequency metadata columns," but I was just following precedent on lines 215 and 210.

File: warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java (hunk 2 above)
Comment (hlgp) [reply warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java]: Considering that we have an addIndexOnlyField method and we don't call it in the equivalent spot at 411, I didn't think it mattered.

**Pull request merged**